src/audio/mux/mux.c

```diff
@@ -399,6 +400,7 @@ static int mux_copy(struct comp_dev *dev)
 	/* align source streams with their respective configurations */
 	list_for_item(clist, &dev->bsource_list) {
 		source = container_of(clist, struct comp_buffer, sink_list);
+		buffer_lock(source, &flags);
 		if (source->source->state == dev->state) {
```

Comment:
Is it possible to return 0 with this lock held at line 412?

Reply:
@dbaluta You're right, else statement with unlock is missing. Will fix that.

Reply:
@tlauda the buffer_unlock should be placed in the ```if(num_sources == 0)``` line.

Like this:

```
if  (num_sources == 0) {
               buffer_unlock(source, flags);
               return 0;
}

Reply:
@dbaluta We need multiple unlocks, so I would need to iterate over all of them, check if they're assigned and unlock them. Better to just unlock immediately if they are not assigned.